experimental/FTheoryTools/src/G4Fluxes/properties.jl

```diff
@@ -0,0 +1,69 @@
+#####################################################
+# 1 Basic properties
+#####################################################
+
+@doc raw"""
+    passes_elementary_quantization_checks(gf::G4Flux)
+
+G4-fluxes are subject to the quantization condition
+[Wit97](@cite) $G_4 + \frac{1}{2} c_2(Y) \in H^{(2,2)}(Y, \mathbb{Z})$.
+It is hard to verify that this condition is met. However,
+we can execute a number of simple consistency checks, by
+verifying that $\int_{Y}{G_4 \wedge [D_1] \wedge [D_2]} \in \mathbb{Z}$
+for any two toric divisors $D_1$, $D_2$. If all of these
+simple consistency checks are met, this method will return
+`true` and otherwise `false`.
+
+It is worth mentioning that currently (July 2024), we only
+support this check for $G_4$-fluxes defined on Weierstrass,
+global Tate and hypersurface models. If this condition is not
+met, this method will return an error.
+
+```jldoctest
+julia> qsm_model = literature_model(arxiv_id = "1903.00009", model_parameters = Dict("k" => 4))
+Hypersurface model over a concrete base
+
+julia> cohomology_ring(ambient_space(qsm_model), check = false);
+
+julia> g4_class = cohomology_class(anticanonical_divisor_class(ambient_space(qsm_model)))^2;
+
+julia> g4 = g4_flux(qsm_model, g4_class, check = false)
+G4-flux candidate lacking elementary quantization checks
+
+julia> passes_elementary_quantization_checks(g4)
+true
+```
+"""
+@attr Bool function passes_elementary_quantization_checks(g4::G4Flux)
+  m = model(g4)
+  @req (m isa WeierstrassModel || m isa GlobalTateModel || m isa HypersurfaceModel) "Elementary quantization checks for  G4-fluxes only supported for Weierstrass, global Tate and hypersurface models"
+  @req base_space(m) isa NormalToricVariety "Elementary quantization checks for G4-flux currently supported only for toric base"
+  @req ambient_space(m) isa NormalToricVariety "Elementary quantization checks for G4-flux currently supported only for toric ambient space"
+
+  # Compute the cohomology class corresponding to the hypersurface equation
+  if m isa WeierstrassModel
+    cl = toric_divisor_class(ambient_space(m), degree(weierstrass_polynomial(m)))
+  end
+  if m isa GlobalTateModel
+    cl = toric_divisor_class(ambient_space(m), degree(tate_polynomial(m)))
+  end
+  if m isa HypersurfaceModel
+    cl = toric_divisor_class(ambient_space(m), degree(hypersurface_equation(m)))
+  end
+  cy = polynomial(cohomology_class(cl))
```

Comment:
This same block of code appears in the calculations of c1, c2, the equality check for fluxes, and here. Perhaps we should make this its own functions and store the result in an attribute.

Reply:
Yeah. Or we add an attribute to Weierstrass and global Tate models that allows us to call hypersurface_equation in those cases to extract the value of `weierstrass_polynomial`, respectively `tate_polynomial`. I will try to remember this for a future PR.